### ISSUE TITLE
Warning: themes:preview is not a zcli command

### ISSUE THREAD

**Issue by javierjulio** [zendesk/zcli#177]
## Expectations

I expected that `npx @zendesk/zcli themes:preview` would work after successfully installing and logging in. I'm migrating from the Ruby gem as it was already deprecated but since 2021 has been reported as getting no further updates. https://github.com/zendesk/zendesk_apps_tools/pull/362

```
npm install @zendesk/zcli
npx zcli login -i
```

## Reality

<img width="637" alt="Screenshot 2023-05-10 at 11 34 12 AM" src="https://github.com/zendesk/zcli/assets/198547/3ffafab4-e3a5-4801-b210-5352f098064a">

## Steps to Reproduce

- `npm install @zendesk/zcli`
- `npx zcli login -i`
- `npx zcli themes:preview` or `npx @zendesk/zcli themes:preview`

### Issue details

- Command: `npx @zendesk/zcli themes:preview`
- Version: 1.0.0-beta.32
- OS: Mac - Ventura 13.3.1 (a)

**Comment by Marco-Pagani** [comment]
I believe that theme previews had not been released yet at the time that you tried to use it. Beta 33 added that functionality, try updating the package and running the command again?

**Comment by Marco-Pagani** [comment]
I apologize for not checking sooner, but it seems that while beta 33 is tagged and 'released' on github, it's not available on npm at the moment. If you want to use it right now you'll likely have to build it from source yourself. Perhaps the team can update this issue when the npm release is available

**Comment by luis-almeida** [comment]
@javierjulio and @Marco-Pagani apologies for the delay. 
Beta 33 should be available now on `npm` as well so you should be able to use the new `zcli themes:preview` command after upgrading the `@zendesk/zcli` package.
Please do let us know if you find any issues or bugs with it. Any feedback would be very much appreciated!

**Comment by Marco-Pagani** [comment]
Thanks for the update @luis-almeida, much appreciated!

I've been using the preview feature for the past week with no issues so far, but I'll be sure to open an issue if I run into anything. Overall though it's a big improvement over ZAT 🙂

**Comment by javierjulio** [comment]
Thanks. I've given this a try but the preview doesn't seem to work right. Either that or the docs are wrong. I expected the defaults where a local server is run using localhost:4567 but that isn't applied. Instead the preview states it's uploaded and to use a URL but then that fails due to CSP checks so CSS and JS are blocked.

Further, live reloading doesn't seem to work either. If I make a change, the console shows a new upload is made but the browser doesn't reload automatically to reflect the change. Although this could be failing due to the CSP errors.

Are we not supposed to load the preview using localhost? I'm prompted to use our subdomain on Zendesk. I expected the former. How can we can get around the CSP errors for local preview?

**Comment by MarkRoodPDQ** [comment]
Sorry, probably a dumb question/issue but when I try previewing it, I get an "Error: expected ";"" error which references a text variable from the manifest.json file. Verified the theme works fine from ZAT. 

Full error:
![image](https://github.com/zendesk/zcli/assets/116672009/e85862c5-7785-4b1c-957f-4baa6e93a489)

**Comment by javierjulio** [comment]
I’m running into that same error with a single field too. The manifest does work with the old ZAT tool as well.

**Comment by luis-almeida** [comment]
@javierjulio the localhost server runs on `http` so, just like with `zat`, one must allow "Mixed content" which is blocked by default.
If you're using Chrome, click the lock icon on the left side of the address bar and select "Site settings". On the "Settings" page, scroll to the "Insecure Content" section and select "Allow".

This is really good feedback! We'll update the docs to make the clearer.

**Comment by luis-almeida** [comment]
@MarkRoodPDQ this might be a bug since it was working fine in `zat`.
We'll need a bit more information here though:
- Are you using SASS in your theme?
- How does your SASS/CSS code look like?
- Can you post a snippet with the text variable definition in the `manifest.json` file?

**Comment by MarkRoodPDQ** [comment]
@luis-almeida, sure thing! Info below:

- Just CSS, no SASS is used
- N/A
-`        {
          "identifier": "hero_title",
          "type": "text",
          "description": "Text above the search field",
          "label": "Title",
          "value": "How can we help?"
        },`

**Comment by javierjulio** [comment]
If it helps, since we had the same error, we are **using SASS** and this was the variable in question.

```
        {
          "identifier": "topbar_link_url_1",
          "type": "text",
          "description": "Specify URL of the link",
          "label": "Topbar link URL",
          "value": "https://jackpocket.com"
        }
```

Terminal error:

```
Error: expected ";".
   ╷
22 │ $topbar_link_url_1: https://jackpocket.com;
   │                          ^
   ╵
  - 22:26  root stylesheet
    at Object.wrapException (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:1237:17)
    at SpanScanner.error$3$length$position (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:69418:15)
    at SpanScanner.expectChar$2$name (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:69500:12)
    at SpanScanner.expectChar$1 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:69503:19)
    at ScssParser0.expectStatementSeparator$1 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:93465:10)
    at ScssParser0.variableDeclarationWithoutNamespace$2 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:96824:13)
    at ScssParser0.variableDeclarationWithoutNamespace$0 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:96831:19)
    at ScssParser0.statements$1 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:93540:35)
    at StylesheetParser_parse_closure0.call$0 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:99986:23)
    at ScssParser0.wrapSpanFormatException$1$1 (/Users/javier/Projects/zendesk-guide-theme/node_modules/sass/sass.dart.js:92441:25)
```

**Comment by javierjulio** [comment]
Same CORS issue after allowing the "Insecure Content" setting.

**Comment by luis-almeida** [comment]
@MarkRoodPDQ, @javierjulio the compilation of the theme variables has changed slightly in `zcli` when compared to `zat`. It's important to keep compatibility of output though for a easier migration though.

How are you using this variable in the CSS or SASS code? Maybe something like this?:

```css
.some-class::before {
  content: "$hero_title";
}
```

I'm particularly interested on wether the variable is within quotes `"$hero_title"`?

Thanks for bearing with us while we address this issue 🙏🏼

**Comment by luis-almeida** [comment]
@MarkRoodPDQ @javierjulio Thanks for confirming so quickly!
This was really helpful. We've started using sass in `themes:preview` to compile theme variables and unfortunately I missed this use case. It was also a mistake to not catch these compilation errors.

I know it's not ideal at all, but until we publish a fix, you should be able to bypass these errors in `zcli` by escaping the problematic characters, e.g. `"How can we help\\?"` or `https\\:\\/\\/jackpocket\\.com`. You'd need to revert that when publishing of course or just use `zat` for now. We should be able to come up with a fix soon now that we've identified the problem so thanks for your help!

**Comment by javierjulio** [comment]
@luis-almeida no worries at all, thank you. I will re-test once the fix is out. Just wanted to provide feedback as we were making a minor update to our template which we hadn't touched in a long time. Thanks again!

**Comment by MarkRoodPDQ** [comment]
Ahh, very excited to try the next beta build!

**Comment by luis-almeida** [comment]
Took a bit longer than expected but we have just published the latest build.
`v1.0.0-beta.34` should fix the reported issue with `preview` and also brings some improvements to template error handling. Note that this version also deprecates the config file in favor of `zcli login -i` (just like in apps):
https://github.com/zendesk/zcli/blob/master/docs/themes.md#zcli-themespreview-themedirectory

Any feedback regarding `preview` would be very much appreciated so we can hopefully close this issue 🙏🏼

**Comment by MarkRoodPDQ** [comment]
Any idea when .34 will hit NPM?

**Comment by MarkRoodPDQ** [comment]
Getting a handful of "is not defined errors" which then causes some elements to not load in ZCLI .34 while it works fine in ZAT. Would it be easier/preferred to create a new issue or keep here?

**Comment by luis-almeida** [comment]
@MarkRoodPDQ sorry about that! 😕
Yes, please create a new issue and include the stack trace. If these are related to the variables in the manifest file, it’d really help to include an example of the variable definition (json) in your manifest file. 

And again, thanks for the feedback!
